master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala

```diff
@@ -1018,12 +1018,16 @@ private[celeborn] class Master(
     // unknown workers will retain in needCheckedWorkerList
     needCheckedWorkerList.removeAll(statusSystem.workers)
     if (shouldResponse) {
+      val excludedList =
```

Comment:
Thanks @ErikFang, Should we consider adding a new method, PbSerDeUtils.toPbPrunedWorkerInfo, to be used in HeartbeatFromApplicationResponse? This could allow us to avoid creating new WorkerInfo objects～

Reply:
good idea, will add it